File: w08/lesson_plans/d03_testing/calculator/calculator.js
var calculator = {

	add: function(numOne, numTwo) {
		firstNum = numOne ? numOne : 0;
		secondNum = numTwo ? numTwo : 0;
		return firstNum + secondNum;
	},

	subtract: function(numOne, numTwo) {
		firstNum = numOne ? numOne : 0;
		secondNum = numTwo ? numTwo : 0;
		return firstNum - secondNum;
	},

	multiply: function(numOne, numTwo) {
		if (!numOne || !numTwo) return 0;
		return numOne * numTwo;
	},

	divide: function(numerator, denominator) {
		if (denominator === 0) return 0;
		return numerator / denominator;
	},

	factorial: function(num) {
		if ( num === 0 ) return 1;
		var result = 1;
		for (var i = num; i > 1; i--) {
			result *= i;
		}
		return result;
	},

	abs: function(num) {
		if ( num * -1 < 0 ) { return num };
		if ( num * -1 > 0 ) { return num * -1}
	}
};

module.exports = calculator

File: w10/lesson_plans/d01_intro_to_ruby/calculator.rb
loop do
puts "Please select a math operation:"
puts "add\nsubtract\nmultiply\ndivide"
puts "Or q to quit"
print ">"
operation = gets.chomp.to_sym

break if operation == :q

puts "Please input two numbers:"
print "Num 1 ->"
num_one = gets.chomp.to_i
print "Num 2 ->"
num_two = gets.chomp.to_i

if operation == :add
	puts "The answer is #{ num_one + num_two }"
elsif operation == :subtract
	puts "The answer is #{ num_one - num_two }"
elsif operation == :multiply
	puts "The answer is #{ num_one * num_two }"
elsif operation == :divide
	puts "The answer is #{ num_one / num_two }" if num_two != 0
	puts "Sorry no zero division in my house" if num_two == 0
end

end